frontend/src/components/Pages/MissionPage/MissionHeader/MissionHeader.tsx

```diff
@@ -27,18 +27,33 @@ interface MissionHeaderProps {
 }
 
 export function MissionHeader({ mission }: MissionHeaderProps) {
+    var remainingTime
     let usedTime = differenceInMinutes(Date.now(), new Date(mission.startTime))
     usedTime = usedTime > 0 ? usedTime : 0
 
-    var remainingTime
-    if (mission.endTime && new Date(mission.endTime).getFullYear() !== 1) {
-        let missionDuration = differenceInMinutes(new Date(mission.endTime), new Date(mission.startTime))
-        let diffNowEndTime = differenceInMinutes(new Date(mission.endTime), Date.now())
-        remainingTime = missionDuration > diffNowEndTime ? diffNowEndTime : missionDuration
-        remainingTime = remainingTime > 0 ? remainingTime : 0
-        remainingTime = remainingTime + ' minutes'
+    let dateTime = mission.estimatedDuration.split('.')
+    const days = dateTime.length === 1 ? 0 : dateTime[0].split(':')[0]
+    const time = dateTime.length === 1 ? dateTime[0].split(':') : dateTime[1].split(':')
+    const estimatedDuration = +time[1] + 60 * (+time[0] + +days * 24)
+
+    if (mission.missionStatus === MissionStatus.Ongoing || mission.missionStatus === MissionStatus.Paused) {
+        if (estimatedDuration && mission.startTime) {
+            remainingTime = estimatedDuration > usedTime ? estimatedDuration - usedTime : 0
+            remainingTime = remainingTime + ' minutes'
+        } else {
+            remainingTime = 'Not available'
+        }
+    } else if (mission.missionStatus === MissionStatus.Pending) {
+        usedTime = 0
+        remainingTime = estimatedDuration + ' minutes'
     } else {
-        remainingTime = 'Not available'
+        if (mission.endTime && mission.startTime) {
+            usedTime = differenceInMinutes(new Date(mission.endTime), new Date(mission.startTime))
+            usedTime = usedTime > 0 ? usedTime : 0
+        } else {
+            usedTime = 0
+        }
+        remainingTime = 'N/A'
```

Comment:
Maybe this can be put into another function, and then called from MissionHeader?
Personally a fan describing the components present in the misison header in this function, and then leave the nitty-gritty details to sub-functions.

Reply:
Agree